Dockerfile

```diff
@@ -0,0 +1,77 @@
+# renovate: datasource=repology depName=alpine_3_16/iptables
+ARG IPTABLES_VERSION=1.8.8-r1
+# renovate: datasource=github-releases depName=kubernetes/dns
+ARG KUBERNETES_DNS_VERSION=1.21.1
+
+
+FROM docker.io/library/alpine:3.16 as iptables-installer
+ARG IPTABLES_VERSION
+ENV IPTABLES_VERSION="${IPTABLES_VERSION:?}"
+# install iptables into a separate root directory so it can be copied into a scratch/distroless image easily with all dependencies
+RUN set -eux; \
+    mkdir -p /rootfs/etc/apk; \
+    cp -r /etc/apk/keys /etc/apk/repositories /rootfs/etc/apk; \
+    apk add \
+        --initdb \
+        --no-cache \
+        --root /rootfs \
+        "iptables=${IPTABLES_VERSION:?}" \
+        ;
+# clean up some directories from /rootfs that are not needed
+RUN set -eux; \
+    rm -fr \
+        /rootfs/dev \
+        /rootfs/etc/apk/keys \
+        /rootfs/proc \
+        /rootfs/tmp \
+        /rootfs/var \
+        ;
+# change default backend from legacy to nf_tables by changing the symlink from xtables-legacy-multi to xtables-nft-multi
+# TODO: find out how to run / innstall https://github.com/kubernetes-sigs/iptables-wrappers in /rootfs (without a shell and without coreutils) instead of changing the symlinks
+RUN set -eux; \
+    find /rootfs/sbin -maxdepth 1 -type l -not -name 'iptables-legacy*' -print | \
+        while read -r FILE; do \
+            if readlink "${FILE:?}" | grep -Fqx xtables-legacy-multi; then \
+                ln -fs xtables-nft-multi "${FILE:?}"; \
+            fi; \
+        done
+# smoke test: print the iptables version and check if the output contains the expected number and the expected backend
+RUN set -eux; \
+    chroot /rootfs iptables --version | \
+        tee -a /dev/stderr | \
+        grep -F "${IPTABLES_VERSION%%-*}" | \
+        grep -Fq nf_tables
+
+
+FROM docker.io/library/golang:1.20.0-alpine3.16 AS node-cache-builder
+WORKDIR /src
+ARG KUBERNETES_DNS_VERSION
+ENV KUBERNETES_DNS_VERSION="${KUBERNETES_DNS_VERSION:?}"
```

Comment:
Isn't that redundant to the line above?

`ARG KUBERNETES_DNS_VERSION` already should provide KUBERNETES_DNS_VERSION as env?